src/java/org/apache/cassandra/audit/AuditLogManager.java

```diff
@@ -0,0 +1,379 @@
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.audit;
+
+import java.util.List;
+import java.util.UUID;
+
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.collect.Lists;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.CQLStatement;
+import org.apache.cassandra.cql3.QueryOptions;
+import org.apache.cassandra.cql3.statements.AlterKeyspaceStatement;
+import org.apache.cassandra.cql3.statements.AlterTypeStatement;
+import org.apache.cassandra.cql3.statements.CFStatement;
+import org.apache.cassandra.cql3.statements.CreateAggregateStatement;
+import org.apache.cassandra.cql3.statements.CreateFunctionStatement;
+import org.apache.cassandra.cql3.statements.CreateKeyspaceStatement;
+import org.apache.cassandra.cql3.statements.CreateTypeStatement;
+import org.apache.cassandra.cql3.statements.DropAggregateStatement;
+import org.apache.cassandra.cql3.statements.DropFunctionStatement;
+import org.apache.cassandra.cql3.statements.DropKeyspaceStatement;
+import org.apache.cassandra.cql3.statements.DropTypeStatement;
+import org.apache.cassandra.cql3.statements.ModificationStatement;
+import org.apache.cassandra.cql3.statements.ParsedStatement;
+import org.apache.cassandra.cql3.statements.SelectStatement;
+import org.apache.cassandra.exceptions.AuthenticationException;
+import org.apache.cassandra.exceptions.UnauthorizedException;
+import org.apache.cassandra.schema.SchemaConstants;
+import org.apache.cassandra.service.QueryState;
+
+public class AuditLogManager
+{
+    public static final Logger logger = LoggerFactory.getLogger(AuditLogManager.class);
+    private static final AuditLogManager instance = new AuditLogManager();
+    private final IAuditLogger auditLogger;
+    private AuditLogManager()
+    {
+        if (isAuditingEnabled())
+        {
+            logger.info("Audit logging is enabled.");
+
+        }
+        else
+        {
+            logger.info("Audit logging is disabled.");
+        }
+        this.auditLogger = getAuditLogger();
+    }
+
+    public static AuditLogManager getInstance()
+    {
+        return instance;
+    }
+
+
+    private IAuditLogger getAuditLogger()
+    {
+        String loggerClassName = DatabaseDescriptor.getAuditLoggingOptions().logger;
+        if (!loggerClassName.contains("."))
+        {
+            loggerClassName = "org.apache.cassandra.audit." + loggerClassName;
+        }
+        logger.info("Logger instance loaded with class : " + loggerClassName);
+        try
+        {
+            return (IAuditLogger) Class.forName(loggerClassName).newInstance();
```

Comment:
Can this code be moved inside util class `FBUtilities.java`? 
Similar to `public static IRoleManager newRoleManager(String className)`

Reply:
Done